lib/ansible/plugins/lookup/first_found.py

```diff
@@ -173,22 +171,21 @@ def run(self, terms, variables, **kwargs):
         else:
             total_search = self._flatten(terms)
 
-        roledir = variables.get('roledir')
         for fn in total_search:
             try:
                 fn = self._templar.template(fn)
-            except (AnsibleUndefinedVariable, UndefinedError) as e:
+            except (AnsibleUndefinedVariable, UndefinedError):
                 continue
 
             if os.path.isabs(fn) and os.path.exists(fn):
                 return [fn]
             else:
-                if roledir is not None:
-                    # check the templates and vars directories too,if they exist
-                    for subdir in ('templates', 'vars', 'files'):
-                        path = self._loader.path_dwim_relative(roledir, subdir, fn)
-                        if os.path.exists(path):
-                            return [path]
+                # check the templates and vars directories too,if they exist
+                #TODO: This should match 'task module' (include_vars/vars, template/templates, etc)
+                for subdir in ('templates', 'vars', 'files'):
```

Comment:
no, this is very wrong as is, we need to select which subdir depending on task action, otherwise include_vars will try to read an template with the same name